lib/mllp/tcp.ex

```diff
@@ -42,9 +44,11 @@ defmodule MLLP.TCP do
   @spec is_closed?(socket :: :gen_tcp.socket()) :: boolean()
   def is_closed?(socket) do
     case recv(socket, _length = 0, _timeout = 1) do
-      {:error, :closed} -> true
-      {:error, :enotconn} -> true
-      _ -> false
+      {:ok, _} -> false
+      {:error, :timeout} -> false
```

Comment:
Final note- let's log here so we can observe it and possibly correlate with reports from the client